src/api/utils/plainbody-decorator.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  BadRequestException,
+  createParamDecorator,
+  ExecutionContext,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import * as getRawBody from 'raw-body';
+
+/**
+ * Extract the raw markdown from the request body and create a new note with it
+ *
+ * Implementation inspired by https://stackoverflow.com/questions/52283713/how-do-i-pass-plain-text-as-my-request-body-using-nestjs
+ */
+export const PlainBody = createParamDecorator(async (_, context: ExecutionContext) => {
+  // we have to check req.readable because of raw-body issue #57
+  // https://github.com/stream-utils/raw-body/issues/57
+  const req = context.switchToHttp().getRequest<import('express').Request>();
+  if (req.readable) {
+    return (await getRawBody(req)).toString().trim();
+  }
+  if (req.get('Content-Type') === 'application/json') {
+    throw new BadRequestException('Body must not be json!');
```

Comment:
Please correct me if I got that wrong: this code prevents clients from uploading `application/json` content but allows all others?

If that's the case, we might want to take an denied-unless-allowed approach instead.

Reply:
We don't want to prevent from uploading something, but if you upload json with Content-Type application/json, NEST parses it and getRawBody will fail. There is an [issue](https://github.com/stream-utils/raw-body/issues/57) which describes that.

Reply:
Maybe "we only accept one content-type" (like `text/plain`) is better than "we accept everything but `application/json`"?

Reply:
changed to text/markdown